src/Kiota.Builder/Refiners/CSharpRefiner.cs

```diff
@@ -42,36 +40,61 @@ public override void Refine(CodeNamespace generatedCode)
             if(currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.Model))
-                currentClass.GetChildElements(true)
-                            .OfType<CodeProperty>()
+                currentClass.Properties
                             .Where(x => x.Type is CodeType propType && propType.TypeDefinition is CodeEnum)
                             .ToList()
                             .ForEach(x => x.Type.IsNullable = true);
             CrawlTree(currentElement, MakeEnumPropertiesNullable);
         }
         private static void AddParsableInheritanceForModelClasses(CodeElement currentElement) {
-            if(currentElement is CodeClass currentClass && currentClass.IsOfKind(CodeClassKind.Model)) {
-                var declaration = currentClass.StartBlock as CodeClass.Declaration;
-                declaration.Implements.Add(new CodeType(currentClass) {
+            if(currentElement is CodeClass currentClass &&
+                currentClass.IsOfKind(CodeClassKind.Model) &&
+                currentClass.StartBlock is CodeClass.Declaration declaration) {
+                declaration.AddImplements(new CodeType {
                     IsExternal = true,
                     Name = $"IParsable",
                 });
-                declaration.Usings.Add(new CodeUsing(currentClass) {
-                    Name = "Microsoft.Kiota.Abstractions.Serialization"
+                (currentClass.Parent is CodeClass parentClass &&
+                parentClass.StartBlock is CodeClass.Declaration parentDeclaration ? 
+                    parentDeclaration :
+                    declaration)
+                    .AddUsings(new CodeUsing {
+                        Name = "Microsoft.Kiota.Abstractions.Serialization"
                 });
             }
             CrawlTree(currentElement, AddParsableInheritanceForModelClasses);
         }
-        private static readonly string[] defaultNamespacesForClasses = new string[] {"System", "System.Collections.Generic", "System.Linq"};
-        private static readonly string[] defaultNamespacesForRequestBuilders = new string[] { "System.Threading.Tasks", "System.IO", "Microsoft.Kiota.Abstractions", "Microsoft.Kiota.Abstractions.Serialization"};
-        private static void AddDefaultImports(CodeElement current) {
-            if(current is CodeClass currentClass) {
-                currentClass.AddUsing(defaultNamespacesForClasses.Select(x => new CodeUsing(currentClass) { Name = x }).ToArray());
-                if(currentClass.IsOfKind(CodeClassKind.RequestBuilder))
-                    currentClass.AddUsing(defaultNamespacesForRequestBuilders.Select(x => new CodeUsing(currentClass) { Name = x }).ToArray());
-            }
-            CrawlTree(current, AddDefaultImports);
-        }
+        private static readonly Tuple<Func<CodeElement, bool>, string, string[]>[] defaultNamespaces = new Tuple<Func<CodeElement, bool>, string, string[]>[] { 
```

Comment:
I really like this approach.
Any reason we can't create a class with the `Func, string and string []` as properties. I think using the tuples might be less verbose but using classes a better form of "documentation".

Reply:
Thanks a lot! Added a record, the code is much cleaner now! 75b6a74 (push pending)